### PR TITLE
Create a custom Rename-File code operation so that we don't show a very unhelpful 'full file added' preview.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/15368

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
Yes.  Zero preview.  Which i think is fine.  It's unclear ot me what a preview coudl show given that the code-action text already says "Rename file to X.cs".

File: src/Workspaces/Core/Portable/CodeActions/Operations/RenameDocumentOperation.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            var oldDocument = solution.GetDocument(_oldDocumentId);
+            var newSolution = solution.RemoveDocument(_oldDocumentId);
+
+            newSolution = newSolution.AddDocument(_newDocumentId, _newFileName, _text, oldDocument.Folders);
+            return workspace.TryApplyChanges(newSolution, progressTracker);
+        }
+
+        public override void Apply(Workspace workspace, CancellationToken cancellationToken)
+        {
+            TryApply(workspace, new ProgressTracker(), cancellationToken);
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/CodeActions/Operations/RenameDocumentOperation.cs]: Is the pattern here supposed to be to toss failure?

File: src/Workspaces/Core/Portable/CodeActions/Operations/RenameDocumentOperation.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            _newDocumentId = newDocumentId;
+            _newFileName = newFileName;
+            _text = text;
+        }
+
+        internal override bool TryApply(Workspace workspace, IProgressTracker progressTracker, CancellationToken cancellationToken)
+        {
+            var solution = workspace.CurrentSolution;
+
+            // currently, document rename is accomplished by a remove followed by an add.
```
Comment (jasonmalinowski) [inline src/Workspaces/Core/Portable/CodeActions/Operations/RenameDocumentOperation.cs]: Just had a random other thought that we've probably had a longstanding bug here: I'm guessing for Team Foundation Version Control we won't correctly apply a rename here so we screw up source control history? Or does something else in the workspace figure this out later?

**Pull request merged**